### PR TITLE
FIX: make deploy does not instantiate the operator

### PR DESCRIPTION
**Pull request by otto-von-bivouac**
Closes: #11

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,11 +2,11 @@
 FROM golang:1.12.5 as builder
 
 WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
+
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
+COPY go.mod go.mod
```
Comment (julio-santiesteban) [inline Dockerfile]: Maybe I'm missing something here, but it looks to me like only the comment was changed here, wasn't it?

**Comment by otto-von-bivouac** [comment]
This was a tricky issue:
```
$ docker run -it --entrypoint /manager kubestone:latest
standard_init_linux.go:207: exec user process caused "no such file or directory"

$ docker run -it --entrypoint /nonexistent kubestone:latest
docker: Error response from daemon: OCI runtime create failed: container_linux.go:344: starting container process caused "exec: \"/nonexistent\": stat /nonexistent: no such file or directory": unknown.

$ docker run -it --entrypoint /bin/sh kubestone:latest
/ # ls -l /manager
-rwxr-xr-x    1 root     root      40603890 Aug  4 05:10 /manager

/ # /manager
/bin/sh: /manager: not found

/ # /lib/ld-musl-x86_64.so.1 /manager
2019-08-04T05:34:54.203Z    ERROR    controller-runtime.client.config    unable to get kubeconfig    {"error": "could not locate a kubeconfig"}
github.com/go-logr/zapr.(*zapLogger).Error
    /go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/client/config.GetConfigOrDie
    /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0-beta.4/pkg/client/config/config.go:143
main.main
    /workspace/main.go:75
runtime.main
    /usr/local/go/src/runtime/proc.go:200
```

The problem is caused by this change: https://github.com/xridge/kubestone/pull/4/files#diff-3254677a7917c6c01f55212f86c57fbfL18-R19

By removing `CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on` the created binary became a dynamically linked one. The builder image had a gnu libc, while the final one was shipped with musl (via alpine). This setup caused the odd `standard_init_linux.go:207: exec user process caused "no such file or directory"` error.

Adding back the necessary flags for the build instructs the compiler to create a statically linked binary.

File: Dockerfile (hunk 1 above)
Comment (otto-von-bivouac) [reply Dockerfile]: This change is not required to fix the build, I just restructured for aesthetic reasons.

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ spec:
         - /manager
         args:
         - --enable-leader-election
-        image: controller:latest
+        image: xridge/kubestone:latest
```
Comment (julio-santiesteban) [inline config/manager/manager.yaml]: Then we should also change the IMG in the Makefile.

Comment (otto-von-bivouac) [reply config/manager/manager.yaml]: good point, addressed it.

**Pull request merged**